src/UI/EditorGroupView.re

```diff
@@ -115,17 +117,122 @@ let make = (~state: State.t, ~windowId: int, ~editorGroup: EditorGroup.t, ()) =>
+          let showLineNumbers =
+            Configuration.getValue(
+              c => c.editorLineNumbers,
+              state.configuration,
+            );
+          let fontSize =
+            Configuration.getValue(
+              c => c.editorFontSize,
+              state.configuration,
+            );
+          let showMinimap =
+            Configuration.getValue(
+              c => c.editorMinimapEnabled,
+              state.configuration,
+            );
+          let maxMinimapCharacters =
+            Configuration.getValue(
+              c => c.editorMinimapMaxColumn,
+              state.configuration,
+            );
+          let matchingPairsEnabled =
+            Selectors.getConfigurationValue(
+              state, Option.value(activeBuffer, ~default=Buffer.empty), c =>
+              c.editorMatchBrackets
+            );
+          let shouldRenderWhitespace =
+            Configuration.getValue(
+              c => c.editorRenderWhitespace,
+              state.configuration,
+            );
+          let shouldRenderIndentGuides =
+            Configuration.getValue(
+              c => c.editorRenderIndentGuides,
+              state.configuration,
+            );
+          let shouldHighlightActiveIndentGuides =
+            Configuration.getValue(
+              c => c.editorHighlightActiveIndentGuide,
+              state.configuration,
+            );
+          let showMinimapSlider =
+            Configuration.getValue(
+              c => c.editorMinimapShowSlider,
+              state.configuration,
+            );
+          let hoverDelay =
+            Configuration.getValue(
+              c => c.editorHoverDelay,
+              state.configuration,
+            )
+            |> Revery.Time.ms;
+          let isHoverEnabled =
+            Configuration.getValue(
+              c => c.editorHoverEnabled,
+              state.configuration,
+            );
+
           <EditorSurface
             isActiveSplit=isActive
-            editorGroup
             metrics
             editor
-            state
-          />
+            activeBuffer
+            onDimensionsChanged
+            onCursorChange
+            onScroll
+            theme
+            rulers
+            showLineNumbers
+            editorFont={state.editorFont}
+            fontSize
+            mode
+            showMinimap
+            maxMinimapCharacters
+            matchingPairsEnabled
+            bufferHighlights={state.bufferHighlights}
+            bufferSyntaxHighlights={state.bufferSyntaxHighlights}
+            diagnostics={state.diagnostics}
+            completions={state.completions}
+            tokenTheme={state.tokenTheme}
+            definition={state.definition}
+            shouldRenderWhitespace
+            showMinimapSlider
+            hoverDelay
+            isHoverEnabled
+            shouldRenderIndentGuides
+            shouldHighlightActiveIndentGuides
+          />;
```

Comment:
A few extra parameters have been added to get rid of the dependency on `State.t`. These are mostly configuration values, fetched above, and some callbacks to dispatch messages.

Reply:
Cool! Thanks for making these dependencies explicit. This is a great step forward to a more-generally-reusable editor component.